packages/blade/src/components/Modal/ModalBody.web.tsx

```diff
@@ -0,0 +1,107 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+import React, { useCallback, useEffect, useState } from 'react';
+import styled from 'styled-components';
+import { modalBodyPadding, scrollOverlayHeight } from './modalTokens';
+import { useModalContext } from './ModalContext';
+import BaseBox from '~components/Box/BaseBox';
+import { assignWithoutSideEffects } from '~src/utils/assignWithoutSideEffects';
+import { MetaConstants, makeMotionTime, makeSize, metaAttribute } from '~utils';
+
+type ModalBodyProps = {
+  children: React.ReactNode;
+};
+
+const OverflowOverlay = styled(BaseBox)<{
+  scrollbarWidth: number;
+  footerHeight: number;
+  showOverlay: boolean;
+}>`
+  position: absolute;
+  left: 0%;
+  right: ${({ scrollbarWidth }) => `${scrollbarWidth}px`};
+  bottom: ${({ footerHeight }) => `${footerHeight}px`};
+  background: linear-gradient(
+    180deg,
+    rgba(255, 255, 255, 0) 0%,
+    rgba(255, 255, 255, 0.6) 47.66%,
+    #ffffff 115.17%
+  );
+  height: ${makeSize(scrollOverlayHeight)};
+  z-index: 999;
+  pointer-events: none;
+  opacity: ${({ showOverlay }) => (showOverlay ? 1 : 0)};
+  transition: ${({ theme }) =>
+    `opacity ${makeMotionTime(theme.motion.duration.moderate)} ${
+      theme.motion.easing.standard.effective
+    }`};
+`;
+
+const _ModalBody = ({ children }: ModalBodyProps): React.ReactElement => {
+  const contentRef = React.useRef<any>(null);
+  const { footerHeight } = useModalContext();
+  const [hasScrollbar, setHasScrollbar] = useState(false);
+  const [scrollbarWidth, setScrollbarWidth] = useState(0);
+  const [scrollEnd, setScrollEnd] = useState(false);
+
+  const checkForScrollbar = useCallback(() => {
+    // Checks if the content has a scrollbar
+    // and sets the width of the scrollbar
+    const element = contentRef.current;
+    if (element) {
+      const hasScrollbar = element.scrollHeight > element.clientHeight;
+      setScrollbarWidth(element.offsetWidth - element.clientWidth);
```

Comment:
Not sure if this works across platforms and OSs, have you confirmed?

Reply:
This doesn't work on macOS when scroll bar is on "auto" mode system-wide. Need to discuss with design if we want to spend more time figuring this out or be okay with the overlap